airflow/ti_deps/deps/ready_to_reschedule.py

```diff
@@ -62,6 +63,9 @@ def _get_dep_statuses(self, ti, session, dep_context):
             .first()
         )
         if not task_reschedule:
+            if is_mapped:
+                yield self._passing_status(reason="The task is mapped and not in reschedule mode")
+                return
```

Comment:
Handling the logic here is the simplest solution among many others I considered. 
`MappedOperator` doesn't serialize sensor attributes.

Reply:
Having `MappedSensor` seems like a better solution.